src/openrct2-ui/windows/Research.cpp

```diff
@@ -154,119 +113,248 @@ static constexpr const StringId ResearchStageNames[] = {
     STR_RESEARCH_STAGE_UNKNOWN,
 };
 
-static void WindowResearchSetPage(WindowBase* w, int32_t page);
-static void WindowResearchSetPressedTab(WindowBase* w);
-static void WindowResearchDrawTabImages(DrawPixelInfo& dpi, WindowBase* w);
-
-WindowBase* WindowResearchOpen()
+class ResearchWindow final : public Window
 {
-    WindowBase* w;
+public:
+    ResearchWindow()
+    {
+        widgets = window_research_page_widgets[WINDOW_RESEARCH_PAGE_DEVELOPMENT];
+        width = WW_DEVELOPMENT;
+        height = WH_DEVELOPMENT;
+    }
 
-    w = WindowBringToFrontByClass(WindowClass::Research);
-    if (w == nullptr)
+    void SetPage(int32_t p)
```

Comment:
Please use descriptive names for variables.

```suggestion
    void SetPage(int32_t newPageIndex)
```

Reply:
Yes, that's better